### PR TITLE
#1607 Fix UBSAN error 'applying non-zero offset ... to null pointer'

### LINKED ISSUE #1607: Fix UBSAN error 'applying non-zero offset ... to null pointer'
```bash
root@59b0e102a283:/# /build/vt/examples/hello_world/objgroup 
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/stl_iterator.h:847:44: runtime error: applying non-zero offset 18446744073709551608 to null pointer
    #0 0x37290d7 in __gnu_cxx::__normal_iterator<char**, std::vector<char*, std::allocator<char*> > >::operator--(int) /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/stl_iterator.h:847:44
    #1 0x370b673 in vt::arguments::ArgConfig::parseArguments[abi:cxx11](CLI::App&, int&, char**&) /vt/src/vt/configs/arguments/args.cc:689:33
    #2 0x370aa2e in vt::arguments::ArgConfig::parse[abi:cxx11](int&, char**&) /vt/src/vt/configs/arguments/args.cc:613:41
    #3 0x2ec2501 in vt::runtime::Runtime::Runtime(int&, char**&, short, bool, int, vt::runtime::eRuntimeInstance) /vt/src/vt/runtime/runtime.cc:156:18
    #4 0x36880e0 in std::_MakeUniq<vt::runtime::Runtime>::__single_object std::make_unique<vt::runtime::Runtime, int&, char**&, short const&, bool&, int&>(int&, char**&, short const&, bool&, int&) /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/unique_ptr.h:857:34
    #5 0x3687ba6 in vt::CollectiveAnyOps<(vt::runtime::eRuntimeInstance)0>::initialize(int&, char**&, short, bool, int*) /vt/src/vt/collective/collective_ops.cc:68:31
    #6 0x364c17d in vt::initialize(int&, char**&, int*) /vt/src/vt/collective/startup.cc:62:10
    #7 0x1ab7025 in main /vt/examples/hello_world/objgroup.cc:60:3
    #8 0x7f01c04100b2 in __libc_start_main /build/glibc-eX1tMB/glibc-2.31/csu/../csu/libc-start.c:308:16
    #9 0x1a9575d in _start (/build/vt/examples/hello_world/objgroup+0x1a9575d)
```

### PR DESCRIPTION
**Pull request by jstrzebonski**
Fixed #1607

### CODE REVIEW

File: src/vt/configs/arguments/args.cc [hunk 1]
```diff
@@ -686,7 +686,7 @@ std::tuple<int, std::string> ArgConfig::parseArguments(CLI::App& app, int& argc,
 
   // Build string-vector and reverse order to parse (CLI quirk)
   std::vector<std::string> args_to_parse;
-  for (auto it = vt_args.end(); it-- != vt_args.begin();) {
+  for (auto it = vt_args.crbegin(); it != vt_args.crend(); ++it) {
```
Comment (cz4rs) [inline src/vt/configs/arguments/args.cc:689]: what about the comment above? this changes the order (it is no longer reversed)

Comment (cz4rs) [reply src/vt/configs/arguments/args.cc:689]: nevermind, I missed the `r` in `crbegin()`

**Review by cz4rs — APPROVED** [review]
Looks good to me.

**Review by PhilMiller — APPROVED** [review]

**Comment by PhilMiller** [comment]
I just did a rough grep over the rest of the repository, and it looks like this is the only instance of this anti-pattern

**Pull request merged**